pkg/pcap/pcaps.go

```diff
@@ -0,0 +1,156 @@
+package pcap
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+//
+// This is a big Pcaps struct holding caches for different types of pcap files
+// to be managed (pcap files per process, per containers and per commands). It
+// would be hard to keep all possible pcap files open forever (as tracee might
+// trace tons of processes, containers or commands).
+//
+// That is what the pcapCache interface is for. It can be implemented by any
+// different way of keeping pcap files and maintain those pcap files cached
+// while they're being used.
+//
+// At the end we have the pcap interface itself. It describes a pcap file being
+// kept opened on behalf of a process, a container or a command.
+//
+// NOTE: Pcaps is not thread safe, should be called from a single routine.
+// TODO: might replace cache_XXX and pcap_XXX files with generics.
+//
+
+//
+// All Pcap Implementations at Once
+//
+
+type Pcaps struct {
+	processes  pcapCache // ProcessPcap Cache (indexed by host_tid)
+	containers pcapCache // ContainersPcap Cache (indexed by container_id)
+	commands   pcapCache // CommandsPcap Cache (indexed by comm)
+}
+
+func NewPcaps(config int, output *os.File) (*Pcaps, error) {
```

Comment:
using the int as a config with flags isn't very readable. This easily could have been a struct with three bool flags and it would be much more readable.

Reply:
I answered Roi on this already. Made it better with const for the flags (and it helps when checking t.config.capture.network != 0 in many parts to see if at least one is enabled, instead of checking multiple bools or another bool that would say something is set).